### PR TITLE
r-universe package list, clearer checkRepoUpToDate, checkup Rprofile

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
- get list of r-universe packages when building lucode2, use that list when building any other package
- checkRepoUpToDate will now throw a helpful error message instead of asking the user when not up to date
- checkup() will check if an Rprofile was found

### CODE REVIEW

File: R/buildLibrary.R [hunk 1]
```diff
@@ -99,6 +99,13 @@ buildLibrary <- function(lib = ".", cran = TRUE, updateType = NULL, gitpush = FA
   checkRepoUpToDate(".", autoCheckRepoUpToDate)
   fixBuildLibraryMergeConflict()
   modifyRproj()
+  if (desc("DESCRIPTION")$get("Package") == "lucode2" &&
+        download.file("https://pik-piam.r-universe.dev/packages", "./rUniversePackages.json.tmp") == 0) {
+      # when building other packages the r-universe badge is added to readme if package name is in this json
+      # rename after successful download to prevent truncated file when download is unsuccessful
+      file.rename("./rUniversePackages.json.tmp", "./inst/extdata/rUniversePackages.json")
+      write("", "./inst/extdata/rUniversePackages.json", append = TRUE) # append newline to prevent warning
+  }
```
Comment (mikapfl) [inline R/buildLibrary.R:108]: Feels weird to have lucode2-specific code in `buildLibrary()`, which I would expect to be package-agnostic. Maybe this is the easiest way to do it right now, but this code could also run in a periodic github action on the lucode2 package or something, right?

File: R/checkRepoUpToDate.R [hunk 2]
```diff
@@ -11,52 +11,46 @@ checkRepoUpToDate <- function(pathToRepo = ".", autoCheckRepoUpToDate = TRUE) {
   }
 
   if (isFALSE(autoCheckRepoUpToDate)) {
-    askUser(NULL)
+    askUser()
     return(invisible(NULL))
```
Comment (mikapfl) [inline R/checkRepoUpToDate.R:15]: I have no idea what `return(invisible(NULL))` does, but I like the sound of it. Not only NULLs, but ghost NULLs :ghost: .

File: R/checkRepoUpToDate.R [hunk 3]
```diff
@@ -11,52 +11,46 @@ checkRepoUpToDate <- function(pathToRepo = ".", autoCheckRepoUpToDate = TRUE) {
...
-      # gert::git_ahead_behind will say we are behind during merge, so cannot use auto check
-      stop()
-    }
-
-    if ("upstream" %in% gert::git_remote_list()[["name"]] &&
-        startsWith(gert::git_remote_info("upstream")[["url"]], "https:pik-piam")) {
-      gert::git_remote_remove("upstream")
-    }
+  # check whether we are merging
+  gitStatus <- system2("git", "status", stdout = TRUE)
```
Comment (mikapfl) [inline R/checkRepoUpToDate.R:23]: The man page says you should use `--porcelain=version`, quote:
>        --porcelain[=<version>]
>           Give the output in an easy-to-parse format for scripts. This is
>           similar to the short output, but will remain stable across Git
>           versions and regardless of user configuration. See below for
>           details.
>
>           The version parameter is used to specify the format version. This is
>           optional and defaults to the original version v1 format.

File: R/package2readme.R [hunk 4]
```diff
@@ -102,12 +102,13 @@ package2readme <- function(package = ".", add = NULL) { # nolint
 
   fillRUniverse <- function(nameOfPackage) {
     # suppress warning about missing trailing newline
-    pikPiamRUniversePackages <- suppressWarnings(readLines("https://pik-piam.r-universe.dev/packages"))
-    pikPiamRUniversePackages <- gsub('[\\", ]', "", pikPiamRUniversePackages)
-    return(ifelse(nameOfPackage %in% pikPiamRUniversePackages,
-                  paste0("[![r-universe](https://pik-piam.r-universe.dev/badges/", nameOfPackage,
-                         ")](https://pik-piam.r-universe.dev/ui#builds)"),
-                  ""))
+    rUniversePackages <- readLines(system.file("extdata", "rUniversePackages.json", package = "lucode2"))
```
Comment (mikapfl) [inline R/package2readme.R:105]: So if people have an old lucode2, the badge will be flicked off again?

**Review by mikapfl — APPROVED** [review]
Looks good to me, the git messages should be much more helpful, thank you!

File: R/checkRepoUpToDate.R (hunk 2 above)
Comment (pfuehrlich-pik) [reply R/checkRepoUpToDate.R:15]: Well this is what you have to do in R to cleanly and explicitly return nothing :see_no_evil:

File: R/package2readme.R (hunk 4 above)
Comment (pfuehrlich-pik) [reply R/package2readme.R:105]: Yes. However, buildLibrary always checks if your lucode2 is outdated, so in practice people are not running buildLibrary with an outdated lucode2 anymore.

File: R/buildLibrary.R (hunk 1 above)
Comment (pfuehrlich-pik) [reply R/buildLibrary.R:108]: There's lucode2 specific code in other places as well already, should I be worried about that? :grimacing: Our checks happen in lucode2::check, which is run by the github action. buildLibrary does both, run the checks and do additional building like this here.

Comment (mikapfl) [reply R/buildLibrary.R:108]: I think, lucode2-specific code in lucode2 is kinda fine. Where I would draw the line would be other-package-specific code in lucode2. :wink:

File: R/checkRepoUpToDate.R (hunk 3 above)
Comment (pfuehrlich-pik) [reply R/checkRepoUpToDate.R:23]: porcelain unfortunately does not show if a merge is in progress once all conflicts have been solved and staged

**Pull request merged**